test/queues.h

```diff
@@ -0,0 +1,57 @@
+#include "greatest.h"
+
+#ifndef DUNST_TEST_QUEUES_H
+#define DUNST_TEST_QUEUES_H
+
+#include <stdbool.h>
+#include <glib.h>
+
+#include "../src/notification.h"
+#include "../src/queues.h"
+
+#define QUEUE_WAIT waiting
+#define QUEUE_DISP displayed
+#define QUEUE_HIST history
+#define QUEUE(q) QUEUE_##q
+
+#define ACT_INIT() queues_init()
+#define ACT_DEINIT() teardown_queues()
+#define ACT_INS(n) queues_notification_insert(n)
+#define ACT_CLS(n) queues_notification_close(n, REASON_UNDEF)
+#define ACT_RUN(data) queues_update(data)
+#define ACT_TMT(a) queues_check_timeouts(a)
+
+#define QUEUE_LEN_ALL(wait, disp, hist) do { \
+        if (wait >= 0) ASSERTm("Waiting is not "   #wait, wait == g_queue_get_length(QUEUE(WAIT))); \
+        if (disp >= 0) ASSERTm("Displayed is not " #disp, disp == g_queue_get_length(QUEUE(DISP))); \
+        if (disp >= 0) ASSERTm("History is not "   #hist, hist == g_queue_get_length(QUEUE(HIST))); \
+        } while (0)
+
+#define QUEUE_IS(q, exp) QUEUE_ISm("QUEUE_IS(" #q "," #exp ")", q, exp)
+#define QUEUE_ISm(msg, q, exp) ASSERTm(msg, (exp == QUEUE(q)))
+#define QUEUE_CONTAINS(q, n) QUEUE_CONTAINSm("QUEUE_CONTAINS(" #q "," #n ")", q, n)
+#define QUEUE_CONTAINSm(msg, q, n) ASSERTm(msg, g_queue_find(QUEUE(q), n))
+
+#define NOT_LAST(n) do {ASSERT_EQm("Notification " #n " should have been deleted.", 1, notification_refcount_get(n)); g_clear_pointer(&n, notification_unref); } while(0)
+
+static inline struct notification *test_notification(const char *name, gint64 timeout)
+{
+        struct notification *n = notification_create();
+
+        if (timeout != -1)
+                n->timeout = S2US(timeout);
```

Comment:
Huh, here's a case I didn't consider: What if timeout is `-1`? It will be multiplied anyway and there are some places in the code we explicitly check for `-1` like [`rules.c`](https://github.com/dunst-project/dunst/blob/master/src/rules.c#L17), need to check if there are breakages with this.

Reply:
`-1` is the default for the notification's timeout:

https://github.com/dunst-project/dunst/blob/d786381cb0d52788de56732fe0f781e4823b6e99/src/notification.c#L312

A timeout of `-1` doesn't interfere with rules at all.

Also in `notification_init` every timeout below 0 will become the default timeout.

https://github.com/dunst-project/dunst/blob/d786381cb0d52788de56732fe0f781e4823b6e99/src/notification.c#L340-L342